### DISCUSSION TITLE
Web Workers (multi-threading)

### DISCUSSION THREAD

**Discussion by genericblueprint** [tauri-apps/tauri#3922] [Q&A] [answered]
Is it possible to execute long running processes in a web worker using Tauri? I could not find any documentation pertaining to multi-threading.

**Comment by JonasKruckenberg** [comment, marked as answer]
Web Workers are of course available in Tauri, it's part of the standard web APIs.

Just know that the Tauri APIs are not available in web workers because of system webview limitations. So you need to route requests to the Rust Core through the main thread.

**Reply by evbo** [reply]
@JonasKruckenberg so let's take this famous js library for example. All it requires is a Boolean `true` and it will use multithreading to parse csv files, avoiding hanging up the browser while it parses your document.
https://www.papaparse.com/docs#config

Are you saying that will work out of the box? Or do I need to call the js library from Rust in order to get the web workers working?

**Reply by JonasKruckenberg** [reply]
From the docs you linked it seems workers are disabled by default anyway, but even with workers enabled the library will just work out of the box. Exactly like it would in a normal website
